packages/schematics/angular/module/index.ts

```diff
@@ -82,18 +152,37 @@ export default function (options: ModuleOptions): Rule {
     options.name = parsedPath.name;
     options.path = parsedPath.path;
 
+    let routingModulePath: Path | undefined;
+    const isLazyLoadedModuleGen = options.route && options.module;
+    if (isLazyLoadedModuleGen) {
+      options.routingScope = RoutingScope.Child;
+      routingModulePath = getRoutingModulePath(host, options);
+    }
+
+    const project = getProject(host, options.project as string);
     const templateSource = apply(url('./files'), [
-      options.routing ? noop() : filter(path => !path.endsWith('-routing.module.ts.template')),
+      options.routing || isLazyLoadedModuleGen && !!routingModulePath
+        ? noop()
+        : filter(path => !path.endsWith('-routing.module.ts.template')),
       applyTemplates({
         ...strings,
         'if-flat': (s: string) => options.flat ? '' : s,
+        lazyRoute: isLazyLoadedModuleGen,
+        routeDeclarationInlined: !routingModulePath,
```

Comment:
Why would we want this? If no routing module is found, we should generate a routing module.

Reply:
Because we would like to stick to the preferred by the user strategy, i.e.: route declarations reside in parent module or in parent routing module. If, for example, the parent module has the declarations inlined, we won't generate a child routing module but rather put the routes in the child module. This is part of the document:

> Depending on whether we’ve inserted the route in the parent routing module or in the parent module [i.e. where is the `RouterModule` declared] we’ll determine if we want to generate a child routing module.

Reply:
Makes sense, thanks